synapse/lib/hiveauth.py

```diff
@@ -1248,7 +1255,22 @@ async def tryPasswd(self, passwd, nexs=True):
             return False
 
         if isinstance(shadow, dict):
-            return await s_passwd.checkShadowV2(passwd=passwd, shadow=shadow)
+            result = await s_passwd.checkShadowV2(passwd=passwd, shadow=shadow)
+
+            if self.auth.policy and (attempts := self.auth.policy['attempts']) > 0:
+                if result:
+                    await self.auth.setUserInfo(self.iden, 'policy:attempts', 0)
+                    return True
+
+                valu = self.info.get('policy:attempts') + 1
+                await self.auth.setUserInfo(self.iden, 'policy:attempts', valu)
+
+                if valu >= attempts:
+                    await self.setLocked(True, logged=nexs)
```

Comment:
If we reach through to `self.nexsroot.cell.setUserLocked()` we get a consistent locking message being logged for a user being locked. Otherwise we'd need to recreate the locking message in this case.